### ISSUE TITLE
How to use waf without local script?

### ISSUE THREAD

**Issue by nowox** [waf-project/waf#2125]
I would like to use waf (and improve it), for the use in embedded projects. We have plenty of projects that need specific build process so the `wscript` is just perfect. However I cannot concede to copy `waf` into all my projects and put it on my Git repositories. 

For my use ~200/300 dependencies + 100 projects = ~400 copies of the waf script. Instead I would like to use a much more common approach using the native executable

`$ waf build` 

where `waf` is the same as `waf-light` script. 

The waf documentation is very well written and very complete. On missing point though is the explanation on why should we build a non-light script from a light one. (waf-light creates waf). What is the purpose of this ? The concept of tar.gz python code inside a python source looks very obscure to me and I would be very glad to get some clues before starting digging in the code (which is well written btw). 

So basically my question is: 

Is it possible to directly use waf as a standard Python module? People do `pip install waf`, they get the `waf` executable out of the box and they just need to write `wscript` files.

**Comment by ita1024** [comment]
The waf file is not a Python module. Make your own wrappers if you deem the standard inapplicable to your case.

**Comment by nowox** [comment]
Thanks for your kindness :)
